ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/Auth.kt

```diff
@@ -128,14 +130,14 @@ public val Auth: ClientPlugin<AuthConfig> = createClientPlugin("Auth", ::AuthCon
 
     on(Send) { originalRequest ->
         val origin = proceed(originalRequest)
-        if (origin.response.status != HttpStatusCode.Unauthorized) return@on origin
+        if (origin.response.status !in pluginConfig.reAuthStatusCodes) return@on origin
         if (origin.request.attributes.contains(AuthCircuitBreaker)) return@on origin
 
         var call = origin
 
         val candidateProviders = HashSet(providers)
 
-        while (call.response.status == HttpStatusCode.Unauthorized) {
+        while (call.response.status in pluginConfig.reAuthStatusCodes) {
```

Comment:
This sounds much better than the list. I've pushed that change.

Reply:
@osipxd 
I like the idea with lambda!

> returning 200 OK with actual status code in a body

Do you mean it could be used like this?
 `isUnauthorized = { it.bodyAsText() == "401" }`
 If yes, then `isUnauthorized` function should be suspend